File: NHunspell/NHunspellExtender2/GrammarEdit.vb
Imports ApiServices.Services
Imports ApiServices.Library

Public Class GrammarEdit

    Public ReadOnly Property StartGrammar As Integer
    Public ReadOnly Property EndGrammar As Integer
    Public ReadOnly Property Sentence As String
    Public ReadOnly Property Result As GrammarBotResult
    Public Property Handled As Boolean
    Public Property Ingored As Boolean

    Public Sub New(startGrammar As Integer, endGrammar As Integer, sentence As String, result As GrammarBotResult)
        Me.StartGrammar = startGrammar
        Me.EndGrammar = endGrammar
        Me.Sentence = sentence
        Me.Result = result
    End Sub

    Public ReadOnly Property HandledOrIgnored As Boolean

        Get
            Return Me.Handled Or Me.Ingored
        End Get

    End Property

End Class

File: NHunspell/NHunspellExtender/GrammarEdit.cs
#if USE_API_SERVICES
using ApiServices.Services;
using ApiServices.Library;

public class GrammarEdit
{
    public int StartGrammar { get; }
    public int EndGrammar { get; }
    public string Sentence { get; }
    public GrammarBotResult Result { get; }
    public bool Handled { get; set; }
    public bool Ingored { get; set; }

    public GrammarEdit(int startGrammar, int endGrammar, string sentence, GrammarBotResult result)
    {
        StartGrammar = startGrammar;
        EndGrammar = endGrammar;
        Sentence = sentence;
        Result = result;
    }

    public bool HandledOrIgnored
    {
        get
        {
            return Handled || Ingored;
        }
    }
}
#endif
